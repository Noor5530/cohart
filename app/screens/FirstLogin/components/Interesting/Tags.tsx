import React, { useEffect, useRef } from "react";
import { Animated, Pressable } from "react-native";
import {  useSelector } from "react-redux";

import { placeHolderColor } from "config/colors";
import AppState from "../../../../models/reducers";

import styles from "./style";

interface ITagsProps {
  setSelectedTopic: (topic: any) => void;
  item: any;
  selectedTopic: any;
  index: number;
}

const Tags = (props: ITagsProps) => {
  const { item, selectedTopic, index } = props;
  // const color = useSharedValue(colors[0]);
  const animation = useRef(new Animated.Value(0));

  const interests = useSelector((state: AppState) => state.user.interests);

  const handleAnimation = () => {
    Animated.timing(animation.current, {
      toValue: 1,
      duration: 800,
      useNativeDriver: false,
    }).start();
  };
  const fadeOut = () => {
    Animated.timing(animation.current, {
      toValue: 0,
      duration: 800,
      useNativeDriver: false,
    }).start();
  };
  const fadeText = animation.current.interpolate({
    inputRange: [0, 1],
    outputRange: [placeHolderColor, "#000000"],
  });
  useEffect(() => {
    interests.map((tag) => {
      if (tag == item.name) {
        handleAnimation();
      }
    });
  }, []);
  const toggleSelection = () => {
    let index = -1;
    index = selectedTopic.findIndex((tag: string) => {
      return tag == item.name;
    });
    if (index !== -1) {
      const data = selectedTopic.filter((tag: string) => {
        return tag !== item.name;
      });

      props.setSelectedTopic(data);
      fadeOut();
    } else {
      const temp = [...props.selectedTopic, item.name];
      props.setSelectedTopic(temp);
      handleAnimation();
    }
  };
  return (
    <Pressable style={styles.row} onPress={toggleSelection}>
      <Animated.Text
        style={[styles.animatedText, { color: fadeText }]}
        key={`id_${index}`}
      >
        {item.name}
      </Animated.Text>
      {item.tag && (
        <Animated.Image
          resizeMode="contain"
          style={[styles.animatedImage, { opacity: animation.current }]}
          source={item.tag}
        />
      )}
    </Pressable>
  );
};

export default Tags;
