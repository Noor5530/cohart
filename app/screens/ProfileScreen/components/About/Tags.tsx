import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { iconColor } from "config/colors";
import { widthPercentageToDP } from "react-native-responsive-screen";

export default function Tags(props: { tag: string }) {
  return (
    <View style={styles.container}>
      <Text style={styles.tag}>{props.tag}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
    borderWidth: 1,
    borderColor: iconColor,
    borderRadius: 25,
    paddingHorizontal: 10,
    marginHorizontal: 5,
    alignItems: "center",
    justifyContent: "center",
    padding: 10,
  },
  tag: {
    fontSize: widthPercentageToDP(2.8),
    fontWeight: "400",
    color: iconColor,
    textAlign: "center",
  },
});
