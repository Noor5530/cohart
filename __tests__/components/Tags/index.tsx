import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import Tags from "../../../app/screens/FirstLogin/components/Interesting/Tags"
jest.useFakeTimers();

jest.mock('react-redux', () => ({
    useDispatch: () => {},
    useSelector: () => ({
      user: {
        interests: []
    },
    }),
  }));

  jest.mock("react-native-paper/src/components/Snackbar", () => "Snackbar");
  jest.mock("react-native-paper/src/components/Button", () => "Button");


test("renders tags correctly", () => {
  const tree = renderer
    .create(
      <Tags
        setSelectedTopic={jest.fn}
        index={0}
        selectedTopic={["a"]}
        item={{ name: "a", tag: { uri: "https://dddd" } }}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

