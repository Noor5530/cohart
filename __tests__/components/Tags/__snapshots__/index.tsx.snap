// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tags correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.2)",
        "fontFamily": "Inter-Regular",
        "fontSize": 40,
        "fontWeight": "bold",
        "paddingBottom": 25,
        "width": 298,
      }
    }
  >
    a
  </Text>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://dddd",
      }
    }
    style={
      Object {
        "height": 50,
        "left": 210,
        "opacity": 0,
        "position": "absolute",
        "top": 10,
        "width": 55,
      }
    }
  />
</View>
`;
